libwacom/libwacom-database.c

```diff
@@ -736,13 +738,17 @@ libwacom_parse_tablet_keyfile(WacomDeviceDatabase *db,
 		device->status_leds = (WacomStatusLEDs *) g_array_free (array, FALSE);
 	}
 
+	success = TRUE;
+
 out:
 	if (path)
 		g_free(path);
 	if (keyfile)
 		g_key_file_free(keyfile);
 	if (error)
 		g_error_free(error);
+	if (!success)
+		device = libwacom_unref(device);
```

Comment:
Might also remove the `g_free (device); device = NULL` statements up on line 602 (the "failed to match" error) since this will take care of it too?

Reply:
fixed, thanks. together with an indentation fix in that same spot.